tiledb/sm/array_schema/attribute.cc

```diff
@@ -209,7 +209,7 @@ Attribute Attribute::deserialize(
   }
 
   std::optional<std::string> enmr_name;
-  if (version >= 19) {
+  if (version >= constants::enumerations_min_format_version) {
```

Comment:
I think we need to introduce a constexpr to handle if its the experimental version or not? Something like what I had in my draft pr for [tile level filters](https://github.com/TileDB-Inc/TileDB/pull/4107/files#diff-e13a4fe0ffc9b5bdaf34d411408ba0dd9ddc39053d9db937c5c5820a9f111a0f).

```
/**
 *
 * Check if the current version is greater than. Handles experimental versions.
 *
 * @param a format version to check
 * @param b format version to compare against
 * @return true if a > b
 */
constexpr bool is_version_greater_than(format_version_t a, format_version_t b) {
  if (is_experimental_build) {
    return (0b10000000000000000000000000000000 | a) > b;
  }
  return a > b;
};
```

Reply:
Yes an no? By definition it can't be a constexpr check since the version is from disk. However, I'm pretty sure constexpr functions can be used non-const?